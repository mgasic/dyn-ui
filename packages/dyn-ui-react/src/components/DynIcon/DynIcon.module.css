.dynIcon {
  --dyn-icon-size: var(--dyn-size-control-md, 1.125rem);
  --dyn-icon-color: var(--dyn-color-text-secondary, currentColor);
  display: inline-flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  vertical-align: middle;
  width: var(--dyn-icon-size);
  height: var(--dyn-icon-size);
  font-size: var(--dyn-icon-size);
  color: var(--dyn-icon-color);
}

.dynIcon svg {
  width: 100%;
  height: 100%;
  fill: currentColor;
  stroke: currentColor;
  display: block;
}

/* Custom React component icons */
.dynIconCustom {
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

/* SVG icons from registry */
.dynIconSvg {
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

/* Fallback text icons */
.dynIconFallback {
  font-size: 0.75em;
  opacity: 0.7;
  font-weight: bold;
  font-family: monospace;
}

.dyn-icon-clickable {
  cursor: pointer;
  border-radius: 4px;
  padding: 2px;
  margin: -2px;
  transition: var(--dyn-animation-transition-all, all 0.2s ease);
}

.dyn-icon-clickable:hover {
  background-color: var(--dyn-color-primary-alpha, rgba(37, 99, 235, 0.16));
  transform: scale(1.04);
}

.dyn-icon-clickable:active {
  transform: scale(0.95);
}

.dyn-icon-clickable:focus-visible {
  outline: 2px solid var(--dyn-color-focus-ring, #3b82f6);
  outline-offset: 2px;
}

/* Size variants mapped to token scale */
.dyn-icon-size-small {
  --dyn-icon-size: var(--dyn-size-control-sm, 0.875rem);
}

.dyn-icon-size-medium {
  --dyn-icon-size: var(--dyn-size-control-md, 1.125rem);
}

.dyn-icon-size-large {
  --dyn-icon-size: var(--dyn-size-control-lg, 1.375rem);
}

/* Visual variants mapped to palette */
.dyn-icon-variant-default {
  --dyn-icon-color: var(--dyn-color-text-secondary, currentColor);
}

.dyn-icon-variant-muted {
  --dyn-icon-color: var(--dyn-color-text-tertiary, currentColor);
}

.dyn-icon-variant-subtle {
  --dyn-icon-color: var(--dyn-color-text-placeholder, currentColor);
}

.dyn-icon-variant-inverse {
  --dyn-icon-color: var(--dyn-color-text-inverse, currentColor);
}

.dyn-icon-variant-accent {
  --dyn-icon-color: var(--dyn-color-primary, currentColor);
}

.fontIcon {
  font-family: var(--dyn-icon-font-family, inherit);
  font-style: normal;
  font-weight: 400;
  line-height: 1;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  speak: none;
}

/* Disabled state */
.disabled {
  opacity: var(--dyn-opacity-disabled, 0.56);
  cursor: not-allowed;
  pointer-events: none;
}

/* Spinning animation */
.spinning {
  animation: iconSpin 1s linear infinite;
}

@keyframes iconSpin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* Reduced motion */
@media (prefers-reduced-motion: reduce) {
  .dyn-icon-clickable {
    transition: none;
  }

  .dyn-icon-clickable:hover {
    transform: none;
  }

  .dyn-icon-clickable:active {
    transform: none;
  }

  .spinning {
    animation: none;
  }
}

/* Print styles - fix deprecated color-adjust */
@media print {
  .dynIcon {
    print-color-adjust: exact;
    -webkit-print-color-adjust: exact;
  }

  .dyn-icon-clickable {
    background: none !important;
    transform: none !important;
  }
}

:global(.dyn-icon) {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  line-height: 1;
}

:global(.dyn-fonts-icon) {
  font-family: var(--dyn-icon-font-family, inherit);
  font-style: normal;
  font-weight: 400;
  line-height: 1;
}
