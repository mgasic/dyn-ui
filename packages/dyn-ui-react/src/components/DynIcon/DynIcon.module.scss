/**
 * DynIcon SCSS Module - Simplified for CSS Modules compatibility
 * Clean selectors that work properly with CSS Module encapsulation
 */

/* Base icon styles */
.dynIcon {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  color: inherit;
  
  svg {
    width: 100%;
    height: 100%;
    fill: currentColor;
    display: block;
  }
}

/* Size variants */
.small {
  width: 14px;
  height: 14px;
}

.medium {
  width: 16px;
  height: 16px;
}

.large {
  width: 18px;
  height: 18px;
}

/* Interactive icons */
.clickable {
  cursor: pointer;
  border-radius: 4px;
  padding: 2px;
  margin: -2px;
  transition: all 0.2s ease;
  
  &:hover {
    background-color: rgba(0, 0, 0, 0.05);
    transform: scale(1.1);
  }
  
  &:active {
    transform: scale(0.95);
  }
  
  &:focus-visible {
    outline: 2px solid #0066cc;
    outline-offset: 2px;
  }
}

/* State colors */
.success {
  color: #059669;
}

.warning {
  color: #d97706;
}

.danger {
  color: #dc2626;
}

.info {
  color: #0891b2;
}

/* Disabled state */
.disabled {
  opacity: 0.5;
  cursor: not-allowed;
  pointer-events: none;
}

/* Spinning animation */
.spinning {
  animation: iconSpin 1s linear infinite;
}

@keyframes iconSpin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* Reduced motion */
@media (prefers-reduced-motion: reduce) {
  .clickable {
    transition: none;
    
    &:hover {
      transform: none;
    }
    
    &:active {
      transform: none;
    }
  }
  
  .spinning {
    animation: none;
  }
}

/* Print styles */
@media print {
  .dynIcon {
    color-adjust: exact;
    -webkit-print-color-adjust: exact;
  }
  
  .clickable {
    background: none !important;
    transform: none !important;
  }
}