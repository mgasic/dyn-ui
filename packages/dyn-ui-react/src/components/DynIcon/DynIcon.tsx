import React from 'react';
import { classNames } from '../../utils/classNames';
import styles from './DynIcon.module.scss';

export interface DynIconProps {
  /** Icon name (string) or React component */
  icon: string | React.ReactNode;
  /** Icon size - can be CSS value or predefined size */
  size?: string | number;
  /** Additional CSS classes */
  className?: string;
  /** Click handler for interactive icons */
  onClick?: () => void;
}

/**
 * DynIcon - Flexible icon component supporting string names and React components
 * Updated to properly use existing SCSS module classes
 */
export const DynIcon: React.FC<DynIconProps> = ({
  icon,
  size,
  className,
  onClick
}) => {
  // Handle React component icons
  if (React.isValidElement(icon)) {
    const iconClasses = classNames(
      styles.dynIcon,
      styles.dynIconCustom,
      {
        [styles.dynIconClickable]: Boolean(onClick),
      },
      className
    );

    const iconStyle: React.CSSProperties = {
      ...(size && {
        fontSize: typeof size === 'number' ? `${size}px` : size,
        width: typeof size === 'number' ? `${size}px` : size,
        height: typeof size === 'number' ? `${size}px` : size,
      }),
    };

    return (
      <span 
        className={iconClasses} 
        style={iconStyle}
        onClick={onClick}
        role={onClick ? 'button' : undefined}
        tabIndex={onClick ? 0 : undefined}
        aria-hidden={!onClick}
      >
        {icon}
      </span>
    );
  }

  // Handle string icon names
  if (typeof icon === 'string') {
    const iconClasses = classNames(
      styles.dynIcon,
      styles.dynIconString,
      {
        [styles.dynIconClickable]: Boolean(onClick),
      },
      className
    );

    const iconStyle: React.CSSProperties = {
      ...(size && {
        fontSize: typeof size === 'number' ? `${size}px` : size,
        width: typeof size === 'number' ? `${size}px` : size,
        height: typeof size === 'number' ? `${size}px` : size,
      }),
    };

    return (
      <i 
        className={iconClasses}
        style={iconStyle}
        data-icon={icon}
        onClick={onClick}
        role={onClick ? 'button' : undefined}
        tabIndex={onClick ? 0 : undefined}
        aria-hidden={!onClick}
        title={icon}
      >
        {/* SVG icons based on name */}
        <svg 
          width="1em" 
          height="1em" 
          viewBox="0 0 16 16" 
          fill="currentColor"
          aria-hidden="true"
        >
          {getIconPath(icon)}
        </svg>
      </i>
    );
  }

  // Return null for invalid icon types
  return null;
};

/**
 * Get SVG path for common icon names
 * This matches the existing icon system from the previous implementation
 */
function getIconPath(iconName: string): React.ReactNode {
  switch (iconName) {
    case 'settings':
    case 'dyn-icon-settings':
      return (
        <>
          <path d="M8 4.754a3.246 3.246 0 1 0 0 6.492 3.246 3.246 0 0 0 0-6.492zM5.754 8a2.246 2.246 0 1 1 4.492 0 2.246 2.246 0 0 1-4.492 0z"/>
          <path d="M9.796 1.343c-.527-1.79-3.065-1.79-3.592 0l-.094.319a.873.873 0 0 1-1.255.52l-.292-.16c-1.64-.892-3.433.902-2.54 2.541l.159.292a.873.873 0 0 1-.52 1.255l-.319.094c-1.79.527-1.79 3.065 0 3.592l.319.094a.873.873 0 0 1 .52 1.255l-.16.292c-.892 1.64.901 3.434 2.541 2.54l.292-.159a.873.873 0 0 1 1.255.52l.094.319c.527 1.79 3.065 1.79 3.592 0l.094-.319a.873.873 0 0 1 1.255-.52l.292.16c1.64.893 3.434-.902 2.54-2.541l-.159-.292a.873.873 0 0 1 .52-1.255l.319-.094c1.79-.527 1.79-3.065 0-3.592l-.319-.094a.873.873 0 0 1-.52-1.255l.16-.292c.893-1.64-.902-3.433-2.541-2.54l-.292.159a.873.873 0 0 1-1.255-.52l-.094-.319zm-2.633.283c.246-.835 1.428-.835 1.674 0l.094.319a1.873 1.873 0 0 0 2.693 1.115l.291-.16c.764-.415 1.6.42 1.184 1.185l-.159.292a1.873 1.873 0 0 0 1.116 2.692l.318.094c.835.246.835 1.428 0 1.674l-.319.094a1.873 1.873 0 0 0-1.115 2.693l.16.291c.415.764-.42 1.6-1.185 1.184l-.291-.159a1.873 1.873 0 0 0-2.693 1.116l-.094.318c-.246.835-1.428.835-1.674 0l-.094-.319a1.873 1.873 0 0 0-2.692-1.115l-.292.16c-.764.415-1.6-.42-1.184-1.185l.159-.291A1.873 1.873 0 0 0 1.945 8.93l-.319-.094c-.835-.246-.835-1.428 0-1.674l.319-.094A1.873 1.873 0 0 0 3.06 4.377l-.16-.292c-.415-.764.42-1.6 1.185-1.184l.292.159a1.873 1.873 0 0 0 2.692-1.115l.094-.319z"/>
        </>
      );
    case 'download':
    case 'dyn-icon-download':
      return (
        <path d="M8.5 1.5A.5.5 0 0 0 8 1a.5.5 0 0 0-.5.5v7.793L5.354 7.146a.5.5 0 1 0-.708.708l3 3a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 9.293V1.5zM3 14.5a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H3z"/>
      );
    case 'ok':
    case 'check':
    case 'dyn-icon-ok':
      return (
        <path d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"/>
      );
    case 'close':
    case 'dyn-icon-close':
      return (
        <path d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8 2.146 2.854z"/>
      );
    case 'warning':
    case 'dyn-icon-warning':
      return (
        <path d="M8.982 1.566a1.13 1.13 0 0 0-1.96 0L.165 13.233c-.457.778.091 1.767.98 1.767h13.713c.889 0 1.438-.99.98-1.767L8.982 1.566zM8 5c.535 0 .954.462.9.995l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 5.995A.905.905 0 0 1 8 5zm.002 6a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"/>
      );
    default:
      // Default icon - a simple circle
      return (
        <circle cx="8" cy="8" r="3" />
      );
  }
}

export default DynIcon;