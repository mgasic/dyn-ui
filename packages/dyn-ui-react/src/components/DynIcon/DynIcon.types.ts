import type { HTMLAttributes, ReactNode } from 'react';
import type { BaseComponentProps } from '../../types';

type DynIconBaseProps = Omit<BaseComponentProps, 'className' | 'id' | 'data-testid'> & {
  className?: string | undefined;
  id?: string | undefined;
  'data-testid'?: string | undefined;
};

export type DynIconTone = 'success' | 'warning' | 'danger' | 'info';
export type DynIconSizeToken = 'small' | 'medium' | 'large';

export interface DynIconProps
  extends Omit<HTMLAttributes<HTMLSpanElement>, 'color'>,
    DynIconBaseProps {
  /** Icon identifier - string (dictionary key, class names) or React node */
  icon?: string | ReactNode;

  /** Icon size token or explicit dimension */
  size?: DynIconSizeToken | number | string;

  /** Semantic tone helper that maps to predefined colors */
  tone?: DynIconTone;

  /** Custom color override */
  color?: string;

  /** Whether the icon should spin */
  spin?: boolean;

  /** Disabled state prevents interaction */
  disabled?: boolean;

  /** Icon content fallback */
  children?: ReactNode;
}

export interface DynIconDefaultProps {
  size: DynIconSizeToken;
  spin: boolean;
  disabled: boolean;
}

export const DYN_ICON_DEFAULT_PROPS: DynIconDefaultProps = {
  size: 'medium',
  spin: false,
  disabled: false,
} as const;

export type IconDictionary = Record<string, string>;

export interface ProcessedIcon {
  baseClass: string;
  iconClass: string;
}
