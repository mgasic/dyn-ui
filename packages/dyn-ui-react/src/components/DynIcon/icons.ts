import * as React from 'react';

// Built-in icon registry with common SVG icons
export const iconRegistry = {
  check: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: React.createElement('path', {
      d: 'M20 6L9 17l-5-5',
      strokeLinecap: 'round',
      strokeLinejoin: 'round'
    })
  }),
  
  close: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('path', {
        key: '1',
        d: 'M18 6L6 18',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('path', {
        key: '2', 
        d: 'M6 6l12 12',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  warning: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('path', {
        key: '1',
        d: 'M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z'
      }),
      React.createElement('line', {
        key: '2',
        x1: '12',
        y1: '9',
        x2: '12',
        y2: '13',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('circle', { key: '3', cx: '12', cy: '17', r: '1' })
    ]
  }),
  
  info: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('circle', { key: '1', cx: '12', cy: '12', r: '10' }),
      React.createElement('path', {
        key: '2',
        d: 'M12 16v-4',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('path', {
        key: '3',
        d: 'M12 8h.01',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  ok: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('circle', { key: '1', cx: '12', cy: '12', r: '10' }),
      React.createElement('path', {
        key: '2',
        d: 'M9 12l2 2 4-4',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  minus: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: React.createElement('line', {
      x1: '5',
      y1: '12',
      x2: '19',
      y2: '12',
      strokeLinecap: 'round',
      strokeLinejoin: 'round'
    })
  }),
  
  plus: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('line', {
        key: '1',
        x1: '12',
        y1: '5',
        x2: '12',
        y2: '19',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('line', {
        key: '2',
        x1: '5',
        y1: '12',
        x2: '19',
        y2: '12',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  download: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('path', {
        key: '1',
        d: 'M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('polyline', {
        key: '2',
        points: '7,10 12,15 17,10',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('line', {
        key: '3',
        x1: '12',
        y1: '15',
        x2: '12',
        y2: '3',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  settings: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('circle', { key: '1', cx: '12', cy: '12', r: '3' }),
      React.createElement('path', {
        key: '2',
        d: 'M12 1v6m0 6v6m11-7h-6m-6 0H1',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      })
    ]
  }),
  
  help: React.createElement('svg', {
    viewBox: '0 0 24 24',
    fill: 'none',
    stroke: 'currentColor',
    strokeWidth: '2',
    children: [
      React.createElement('circle', { key: '1', cx: '12', cy: '12', r: '10' }),
      React.createElement('path', {
        key: '2',
        d: 'M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3',
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
      }),
      React.createElement('circle', { key: '3', cx: '12', cy: '17', r: '1' })
    ]
  }),
};

export type IconName = keyof typeof iconRegistry;

export const getIcon = (name: string): React.ReactElement | null => {
  return iconRegistry[name as IconName] || null;
};