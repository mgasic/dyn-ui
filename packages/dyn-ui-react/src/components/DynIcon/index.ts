// Standardized exports for DynIcon
export { DynIcon } from './DynIcon';
export { default } from './DynIcon';
export type { DynIconProps } from './DynIcon.types';

export type { IconDictionary } from './DynIcon.types';
export type { IconDictionaryContext } from '../../providers/IconDictionaryProvider';
export * from './icons';
