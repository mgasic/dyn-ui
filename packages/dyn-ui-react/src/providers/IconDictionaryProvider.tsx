import { createContext, useContext, useEffect, useState } from 'react';
import { IconDictionary } from '../types/icon.types';

interface IconDictionaryContextType {
  [key: string]: string;
}

export const IconDictionaryContext = createContext<IconDictionaryContextType>({});

interface Props {
  children: React.ReactNode;
  customDictionary?: IconDictionary;
}

export const IconDictionaryProvider: React.FC<Props> = ({ children, customDictionary }: Props) => {
  const [dictionary, setDictionary] = useState<IconDictionaryContextType>({});

  useEffect(() => {
    // Load default DYN UI icons + custom dictionary
    const defaultIcons: IconDictionary = {
      'user': 'dyn-icon-user',
      'home': 'dyn-icon-home',
      'settings': 'dyn-icon-settings',
      'ok': 'dyn-icon-ok',
      'close': 'dyn-icon-close',
      'warning': 'dyn-icon-warning',
      'minus': 'dyn-icon-minus',
      'plus': 'dyn-icon-plus',
      'search': 'dyn-icon-search',
      'edit': 'dyn-icon-edit',
      'delete': 'dyn-icon-delete',
      'arrow-up': 'dyn-icon-arrow-up',
      'arrow-down': 'dyn-icon-arrow-down',
      'arrow-left': 'dyn-icon-arrow-left',
      'arrow-right': 'dyn-icon-arrow-right',
      'menu': 'dyn-icon-menu',
      'info': 'dyn-icon-info',
      'calendar': 'dyn-icon-calendar',
      'clock': 'dyn-icon-clock',
      'mail': 'dyn-icon-mail'
      // Add more default mappings as needed
    };

    setDictionary({
      ...defaultIcons,
      ...customDictionary
    });
  }, [customDictionary]);

  return (
    <IconDictionaryContext.Provider value={dictionary}>
      {children}
    </IconDictionaryContext.Provider>
  );
};
