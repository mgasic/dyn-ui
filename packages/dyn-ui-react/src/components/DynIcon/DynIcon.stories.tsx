import type { Meta, StoryObj } from '@storybook/react';
import { React } from 'react';
import { DynIcon } from './DynIcon';
import { ThemeProvider } from '../../theme/ThemeProvider';
import { IconDictionaryProvider } from '../../contexts/IconDictionaryContext';

const meta: Meta<typeof DynIcon> = {
  title: 'Components/DynIcon',
  component: DynIcon,
  decorators: [(S) => <ThemeProvider initialTheme="light"><S/></ThemeProvider>],
  decorators: [(S) => <IconDictionaryProvider><S/></IconDictionaryProvider>],
};
export default meta;
type Story = StoryObj<typeof DynIcon>;
export const Playground: Story = { args: {} };
