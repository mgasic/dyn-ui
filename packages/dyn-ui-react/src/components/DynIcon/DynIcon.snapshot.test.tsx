import { render } from '@testing-library/react';
import React from 'react';
import { describe, it, expect } from 'vitest';
import DynIcon from './DynIcon';
import { IconDictionaryProvider } from '../../providers/IconDictionaryProvider';

const renderSnapshot = (element: React.ReactElement) =>
  render(<IconDictionaryProvider>{element}</IconDictionaryProvider>);

describe('DynIcon snapshots', () => {
  it('matches default render', () => {
    const { container } = renderSnapshot(<DynIcon icon="ok" />);
    expect(container).toMatchSnapshot();
  });

  it('matches registry icon render', () => {
    const { container } = renderSnapshot(<DynIcon icon="check" size="large" />);
    expect(container).toMatchSnapshot();
  });
});
