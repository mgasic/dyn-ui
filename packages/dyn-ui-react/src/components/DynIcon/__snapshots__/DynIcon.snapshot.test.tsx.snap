// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DynIcon snapshots > matches default render 1`] = `
<div>
  <span
    aria-hidden="true"
    class="_dynIcon_398cbd _dyn-icon-size-medium_398cbd dyn-icon dyn-icon-ok"
    role="img"
  />
</div>
`;

exports[`DynIcon snapshots > matches registry icon render 1`] = `
<div>
  <span
    aria-hidden="true"
    class="_dynIcon_398cbd _dyn-icon-size-large_398cbd"
    role="img"
  >
    <span
      aria-hidden="true"
      class="_dynIconSvg_398cbd"
    >
      <svg
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6L9 17l-5-5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </span>
  </span>
</div>
`;
